cmd/nginx-ingress/flags.go

```diff
@@ -0,0 +1,419 @@
+}
+
+// validatePort makes sure a given port is inside the valid port range for its usage
+func validatePort(port int) error {
+	if port < 1024 || port > 65535 {
+		return fmt.Errorf("port outside of valid port range [1024 - 65535]: %v", port)
+	}
+	return nil
+}
+
+const appProtectLogLevelDefault = "fatal"
+
+// validateAppProtectLogLevel makes sure a given logLevel is one of the allowed values
+func validateAppProtectLogLevel(logLevel string) error {
+	switch strings.ToLower(logLevel) {
+	case
+		"fatal",
+		"error",
+		"warn",
+		"info",
+		"debug",
+		"trace":
+		return nil
+	}
+	return fmt.Errorf("invalid App Protect log level: %v", logLevel)
+}
+
+// parseNginxStatusAllowCIDRs converts a comma separated CIDR/IP address string into an array of CIDR/IP addresses.
+// It returns an array of the valid CIDR/IP addresses or an error if given an invalid address.
+func parseNginxStatusAllowCIDRs(input string) (cidrs []string, err error) {
+	cidrsArray := strings.Split(input, ",")
+	for _, cidr := range cidrsArray {
+		trimmedCidr := strings.TrimSpace(cidr)
+		err := validateCIDRorIP(trimmedCidr)
+		if err != nil {
+			return cidrs, err
+		}
+		cidrs = append(cidrs, trimmedCidr)
+	}
+	return cidrs, nil
+}
+
+// validateCIDRorIP makes sure a given string is either a valid CIDR block or IP address.
+// It an error if it is not valid.
+func validateCIDRorIP(cidr string) error {
+	if cidr == "" {
+		return fmt.Errorf("invalid CIDR address: an empty string is an invalid CIDR block or IP address")
+	}
+	_, _, err := net.ParseCIDR(cidr)
+	if err == nil {
+		return nil
+	}
+	ip := net.ParseIP(cidr)
+	if ip == nil {
+		return fmt.Errorf("invalid IP address: %v", cidr)
+	}
+	return nil
+}
+
+const (
+	locationFmt    = `/[^\s{};]*`
+	locationErrMsg = "must start with / and must not include any whitespace character, `{`, `}` or `;`"
+)
+
+var locationRegexp = regexp.MustCompile("^" + locationFmt + "$")
+
+func validateLocation(location string) error {
+	if location == "" || location == "/" {
+		return fmt.Errorf("invalid location format: '%v' is an invalid location", location)
+	}
+	if !locationRegexp.MatchString(location) {
+		msg := validation.RegexError(locationErrMsg, locationFmt, "/path", "/path/subpath-123")
+		return fmt.Errorf("invalid location format: %v", msg)
+	}
+	return nil
+}
+
+func getBuildInfo() (string, string) {
+	info, ok := debug.ReadBuildInfo()
+	if !ok {
+		return "", ""
+	}
+	for _, kv := range info.Settings {
+		switch kv.Key {
+		case "vcs.revision":
+			commitHash = kv.Value
+		case "vcs.time":
+			commitTime = kv.Value
+		case "vcs.modified":
+			dirtyBuild = kv.Value == "true"
+		}
+	}
+	binaryInfo := fmt.Sprintf("Commit=%v Date=%v DirtyState=%v Arch=%v/%v Go=%v", commitHash, commitTime, dirtyBuild, runtime.GOOS, runtime.GOARCH, runtime.Version())
+	versionInfo := fmt.Sprintf("Version=%v", version)
+
+	if *versionFlag {
+		fmt.Println(versionInfo)
+		fmt.Println(binaryInfo)
+		os.Exit(0)
+	}
+	return versionInfo, binaryInfo
+}
```

Comment:
I feel like this shouldn't be in `flags.go`? Is there any way that we can move it out? Probably the build info read from the binary could go in `utils.go`.

Reply:
Yeah, I didn't feel like it made a lot of sense here either, the only reason I left it here in because of the `versionFlag`. I've moved it (without the `versionFlag` part) to a new `utils.go` file, call this function from `main.go` and process the versionFlag part only in `flags.go`. Let me know what you think.